### PR TITLE
Clean workflows

### PR DESCRIPTION
**Pull request by KPreisner**
Clean up workflows

- Separate linter and build & publish workflows
- Change run strategy to on release for build & publish and generate docs workflows
- Light renaming of workflows and jobs
- Linter workflow currently does nothing, so the entire workflow file is commented out so it doesn't waste github action runtime minutes

### CODE REVIEW

File: .github/workflows/lint-check.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+#       #   uses: actions/checkout@v4
+
+#       # - name: Set Up Python
+#       #   uses: actions/setup-python@v4
+#       #   with:
+#       #     python-version: '3.10'
+
+#       - name: Linting
+#         # TODO: uncomment above steps for checking out the code and setting up python when Linting is implemented.
+#         run: echo "black, isort, flake8, mypy still to come..."
```
Comment (sangaline) [inline .github/workflows/lint-check.yml]: I'd just delete this, it's still in the git history if you need it later and it only adds noise right now.

Comment (KPreisner) [reply .github/workflows/lint-check.yml]: ok can do

Comment (KPreisner) [reply .github/workflows/lint-check.yml]: deleted [9e99969](https://github.com/Sindri-Labs/sindri-python/pull/14/commits/9e99969e1e146a8c44e5d51cb91897561b2254fa)

**Review by sangaline — APPROVED** [review]
lgtm

**Pull request merged**